src/webgpu/idl/javascript.spec.ts

```diff
@@ -0,0 +1,339 @@
+import {
+  DeviceSelectionDescriptor,
+  GPUTest,
+  GPUTestSubcaseBatchState,
+  initUncanonicalizedDeviceDescriptor,
+} from '../gpu_test.js';
+import { CanonicalDeviceDescriptor, DescriptorModifier } from '../util/device_pool.js';
+
+function addAllFeatures(adapter: GPUAdapter, desc: CanonicalDeviceDescriptor | undefined) {
+  const descWithMaxLimits: CanonicalDeviceDescriptor = {
+    defaultQueue: {},
+    ...desc,
+    requiredFeatures: [...adapter.features] as GPUFeatureName[],
+    requiredLimits: {},
+  };
+  return descWithMaxLimits;
+}
+
+/**
+ * Used to request a device with all the max limits of the adapter.
+ */
+class AllFeaturesGPUTestSubcaseBatchState extends GPUTestSubcaseBatchState {
+  override requestDeviceWithRequiredParametersOrSkip(
+    descriptor: DeviceSelectionDescriptor,
+    descriptorModifier?: DescriptorModifier
+  ): void {
+    const mod: DescriptorModifier = {
+      descriptorModifier(adapter: GPUAdapter, desc: CanonicalDeviceDescriptor | undefined) {
+        desc = descriptorModifier?.descriptorModifier
+          ? descriptorModifier.descriptorModifier(adapter, desc)
+          : desc;
+        return addAllFeatures(adapter, desc);
+      },
+      keyModifier(baseKey: string) {
+        return `${baseKey}:AllFeaturesTest`;
+      },
+    };
+    super.requestDeviceWithRequiredParametersOrSkip(
+      initUncanonicalizedDeviceDescriptor(descriptor),
+      mod
+    );
+  }
+}
+
+/**
+ * A Test that requests all the max limits from the adapter on the device.
+ */
+class AllFeaturesTest extends GPUTest {
+  public static override MakeSharedState(
+    recorder: TestCaseRecorder,
+    params: TestParams
+  ): GPUTestSubcaseBatchState {
+    return new AllFeaturesGPUTestSubcaseBatchState(recorder, params);
+  }
+}
+
+const kResourceInfo = {
+  buffer: {
+    create(t: GPUTest) {
+      return t.createBufferTracked({ size: 16, usage: GPUBufferUsage.UNIFORM });
+    },
+    requiredKeys: [
+      'destroy',
+      'getMappedRange',
+      'label',
+      'mapAsync',
+      'mapState',
+      'size',
+      'unmap',
+      'usage',
+    ],
+  },
+  texture: {
+    create(t: GPUTest) {
+      return t.createTextureTracked({
+        size: [2, 3],
+        format: 'r8unorm',
+        usage: GPUTextureUsage.TEXTURE_BINDING,
+      });
+    },
+    requiredKeys: [
+      'createView',
+      'depthOrArrayLayers',
+      'destroy',
+      'dimension',
+      'format',
+      'height',
+      'label',
+      'mipLevelCount',
+      'sampleCount',
+      'usage',
+      'width',
+    ],
+  },
+  querySet: {
+    create(t: GPUTest) {
+      return t.createQuerySetTracked({
+        type: 'occlusion',
+        count: 2,
+      });
+    },
+    requiredKeys: ['count', 'destroy', 'label', 'type'],
+  },
+  adapter: {
+    create(t: GPUTest) {
+      return t.adapter;
+    },
+    requiredKeys: [
+      'featureLevel',
+      'features',
+      'info',
+      'isCompatibilityMode',
```

Comment:
Oh, this shouldn't be there anymore though - noticed it failed in stable. Maybe some of these keys should be optional? (featureLevel, isCompatibilityMode)

Reply:
fixed